File: briareos-master/dev/project/engine/modules/xss_detector.py
# Pipeline Module

from briareos.utils.modules import *

from http_object import HttpObject


class XSSDetector(Module):
    name = "XSS Detector"
    description = ""
    input_type = HttpObject
    output_type = str

    def process(self, packet, http_object):
        return "XSS Detector output"

File: jbriareos-master/Tadeu_Freitas-jbriareos-c16b99c2c2d9/src/main/java/engine_modules/XssDetector.java
package engine_modules;

import core.JMSInterface;
import engine_modules_helpers.IOData;
import engine_modules_helpers.IOType;
import engine_modules_helpers.ModuleIO;
import engine_modules_helpers.ModuleType;
import utils.BPacket;
import utils.Common;

public class XssDetector extends EngineModule {
    public XssDetector() {
        super(ModuleType.XSSDetector);
    }

    @Override
    public IOType getInputType() {
        return IOType.HTTPObject;
    }

    @Override
    public IOType getOutputType() {
        return IOType.String;
    }

    @Override
    public Common.InputMode getInputMode() {
        return Common.InputMode.SingleInput;
    }

    @Override
    public ModuleIO process(BPacket packet, ModuleIO data, JMSInterface jmsInterface) {
        return new ModuleIO(IOType.String, new IOData.IOString("XSS Detector output"));
    }
}
